framework/src/media/audio/audio_manager.c

```diff
@@ -1025,85 +1022,75 @@ uint32_t get_output_bytes_to_frame(unsigned int bytes)
-	}
-	medvdbg("Max volume: %d, Current Volume : %d card path : %s\n", g_audio_in_cards[g_actual_audio_in_card_id].config.max_volume, g_audio_in_cards[g_actual_audio_in_card_id].config.volume, g_audio_in_cards[g_actual_audio_in_card_id].card_path);
+	/* get the gain of the current card first */
+	ret = get_audio_volume(fd, &(g_audio_in_cards[g_actual_audio_in_card_id].config), INPUT);
+	medvdbg("ret = %d, Max gain: %d, Current gain : %d card path : %s\n", ret, g_audio_in_cards[g_actual_audio_in_card_id].config.max_volume, g_audio_in_cards[g_actual_audio_in_card_id].config.volume, g_audio_in_cards[g_actual_audio_in_card_id].card_path);
+
+	*gain = g_audio_in_cards[g_actual_audio_in_card_id].config.volume;
 
 	close(fd);
 
-	return (int)g_audio_in_cards[g_actual_audio_in_card_id].config.volume;
-#endif
-	return 0;
+	return ret;
 }
 
-int get_output_audio_volume(void)
+audio_manager_result_t get_output_audio_volume(uint8_t *volume)
 {
-#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
-	int ret;
+	int ret = AUDIO_MANAGER_SUCCESS;
 	int fd;
 
+	if (volume == NULL) {
+		return AUDIO_MANAGER_INVALID_PARAM;
+	}
+
 	fd = open(g_audio_out_cards[g_actual_audio_out_card_id].card_path, O_RDONLY);
 	if (fd < 0) {
 		meddbg("Failed to open audio driver, path : %s errno : %d\n", g_audio_out_cards[g_actual_audio_out_card_id].card_path, errno);
 		return AUDIO_MANAGER_CARD_NOT_READY;
 	}
 
-	/* get volume of current card first */
-	if ((ret = get_audio_volume(fd, &(g_audio_out_cards[g_actual_audio_out_card_id].config), OUTPUT)) != AUDIO_MANAGER_SUCCESS) {
-		meddbg("Get volume failed card path : %s card_type : %d ret : %d\n", g_audio_out_cards[g_actual_audio_out_card_id].card_path, INPUT, ret);
-		close(fd);
-		return ret;
-	}
-	medvdbg("Max volume: %d, Current Volume : %d card path : %s\n", g_audio_out_cards[g_actual_audio_out_card_id].config.max_volume, g_audio_out_cards[g_actual_audio_out_card_id].config.volume, g_audio_out_cards[g_actual_audio_out_card_id].card_path);
+	/* get the volume of the current card first */
+	ret = get_audio_volume(fd, &(g_audio_out_cards[g_actual_audio_out_card_id].config), OUTPUT);
+	medvdbg("ret = %d, Max volume: %d, Current Volume : %d card path : %s\n", ret, g_audio_out_cards[g_actual_audio_out_card_id].config.max_volume, g_audio_out_cards[g_actual_audio_out_card_id].config.volume, g_audio_out_cards[g_actual_audio_out_card_id].card_path);
+
+	*volume = g_audio_out_cards[g_actual_audio_out_card_id].config.volume;
 
 	close(fd);
 
-	return (int)g_audio_out_cards[g_actual_audio_out_card_id].config.volume;
-#endif
-	return 0;
+	return ret;
 }
 
-audio_manager_result_t set_input_audio_volume(uint8_t volume)
+audio_manager_result_t set_input_audio_gain(uint8_t gain)
 {
-#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
-	if (g_audio_in_cards[g_actual_audio_in_card_id].config.volume == volume) {
-		return AUDIO_MANAGER_SUCCESS;
-	}
-
-	return set_audio_volume(INPUT, volume);
-#endif
-	return AUDIO_MANAGER_SUCCESS;
+	return set_audio_volume(INPUT, gain);
```

Comment:
set_audio_volume doens't include logic regarding exclude. please check it.

Reply:
I updated (classified) the return case of ioctl in set_audio_volume related to the exclude config.